src/modules/demo_playback.rs

```diff
@@ -134,16 +153,28 @@ fn find_demos(prefix: PathBuf) -> Result<Vec<(PathBuf, usize)>, String> {
         .filter_map(Result::ok)
         .map(|entry| entry.path())
         .filter(|path| path.extension().map(|ext| ext == "dem").unwrap_or(false))
-        .filter_map(|path| {
+        .enumerate()
+        .filter_map(|(index, path)| {
             path.file_stem()
                 .and_then(OsStr::to_str)
-                .filter(|name| name.starts_with(&name_prefix))
-                .and_then(|name| name[name_prefix.len()..].parse::<usize>().ok())
```

Comment:
Please don't add it into this function. The expression is already way too complicated, the last thing it needs is more special cases. Even just writing out a separate function with the same functionality would be better.

Reply:
👍

Reply:
The `find_demos()` now has one for folder as requested 👍 I still have play_folder depending on play_run as it would be easier to maintain, I think.